pyanaconda/modules/common/structures/kickstart.py

```diff
@@ -0,0 +1,138 @@
+#
+# DBus structures for kickstart.
+#
+# Copyright (C) 2019  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from pyanaconda.dbus.structure import DBusData
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+
+__all__ = ["KickstartMessage", "KickstartReport"]
+
+
+class KickstartMessage(DBusData):
+    """The kickstart message."""
+
+    def __init__(self):
+        self._module_name = ""
+        self._file_name = ""
+        self._line_number = 0
+        self._message = ""
+
+    @property
+    def module_name(self) -> Str:
+        """Name of the DBus module.
+
+        :return: a DBus name
+        """
+        return self._module_name
+
+    @module_name.setter
+    def module_name(self, value: Str):
+        self._module_name = value
+
+    @property
+    def file_name(self) -> Str:
+        """Name of the file.
+
+        :return: a file name
+        """
+        return self._file_name
+
+    @file_name.setter
+    def file_name(self, value: Str):
+        self._file_name = value
+
+    @property
+    def line_number(self) -> Int:
+        """Number of the line.
+
+        :return: a number
+        """
+        return self._line_number
+
+    @line_number.setter
+    def line_number(self, value: Int):
+        self._line_number = value
+
+    @property
+    def message(self) -> Str:
+        """Translated message.
+
+        :return: a string
+        """
+        return self._message
+
+    @message.setter
+    def message(self, value: Str):
+        self._message = value
+
+    @classmethod
+    def for_error(cls, e):
+        """Create a new message for a kickstart error.
+
+        :param e: an instance of KickstartError
+        :return: an instance of KickstartMessage
+        """
+        data = cls()
+        data.message = e.message
+        data.line_number = e.lineno
+        return data
```

Comment:
This is just a factory method. We use something similar in the DBus structure `Requirement`, where we use `for_package` and `for_group` to create requirements for a package or a group. I am not sure how to explain it better.

Reply:
Just to be clear, does this mean that the only way how the class should be created is to use the `for_error()` method? If yes than this can be specified in the constructor doc string.

--------------------------------

Also I know we are using it elsewhere but I'm thinking if `for_*` prefix is used somewhere else than Anaconda project. Is it a convention for constructors? I know convention `new_from_error()` or similar but this does not seems familiar to me. I won't block this PR because of the name but it could be an interesting for other PR.

Reply:
No, the kickstart message can be created the usual way. This is just a shortcut for the kickstart exceptions.

I couldn't find any naming conventions for constructors. I like the `for_` prefix, because it is easy to read in the code. For example, `KickstartMessage.for_error(e)` literally means create a new instance of the class `KickstartMessage` for the error `e`. Also it cannot be confused with a usual method. However, we can discuss it with the team and change it in another PR.